pennylane/devices/qubit/apply_operation.py

```diff
@@ -228,9 +229,27 @@ def _(op: type_op, state):
         [1., 0.]], requires_grad=True)
 
     """
+    if sp.sparse.issparse(op.matrix()):
+        return apply_operation_csr_matrix(op, state, is_state_batched)
     return _apply_operation_default(op, state, is_state_batched, debugger)
 
 
+def apply_operation_csr_matrix(op, state, is_state_batched: bool = False):
+    """The csr_matrix specialized version apply operation."""
+    # Calculate the num wires by state shape
+    if sp.sparse.issparse(state):  # Then the first is batch and the second is state dim
+        len_state = state.shape[1]
+        num_wires = int(np.log2(len_state))
+        return state @ op.matrix(wire_order=range(num_wires)).T
+    # Then state is numpy array, should have been stored in tensor version
+    original_shape = math.shape(state)
+    num_wires = len(original_shape) - is_state_batched
+    full_state = math.reshape(state, [-1, 2**num_wires])  # expected: [batch_size, 2**num_wires]
+    state_opT = full_state @ op.matrix(wire_order=range(num_wires)).T
+    state_reshaped = math.reshape(state_opT, original_shape)
+    return state_reshaped
```

Comment:
@AmintorDusko  Here's the path for nparray state to be apply by sparse op

Reply:
Once again. This is great, thanks!